sdk/cosmosdb/cosmos/src/client/Item/Item.ts

```diff
@@ -41,10 +41,10 @@ export class Item {
   constructor(
     public readonly container: Container,
     public readonly id: string,
-    partitionKey: PartitionKey,
+    partitionKey: PartitionKey | undefined,
```

Comment:
We may want to have a constructor overload with no partitionKey instead of making this optional. This is important in the scenario where the constructor is called with no partition key so that we don't force callers to pass `undefined` 

```ts
new Item(container, id, undefined, myContext);
//vs with the overload
new Item(container, id, myContext);
```

Reply:
Thanks for the great suggestion. Luckily this is an internal api, I will make `partitionKey` last parameter and mark it optional.